site/source/docs/compiling/WebAssembly.rst

```diff
@@ -22,17 +22,22 @@ WebAssembly is emitted by default, without the need for any special flags.
 
   emcc [..args..] -s WASM=0
 
-.. note:: Emscripten's WebAssembly support depends on `Binaryen <https://github.com/WebAssembly/binaryen>`_, which will be automatically fetched and built for you (you may see logging about that, the first time you compile to WebAssembly).
-.. note:: The ``WASM``, ``BINARYEN*``, etc. options only matter when compiling to your final executable. In other words, the same .o files are used for both asm.js and WebAssembly. Only when linking them and compiling to asm.js or WebAssembly do you need to specify WebAssembly if you want that. That means that it is easy to build your project to both asm.js and WebAssembly.
+.. note:: Emscripten's WebAssembly support depends on `Binaryen <https://github.com/WebAssembly/binaryen>`_, which is provided by the emsdk (if you don't use the emsdk, you need to build it and set it up in your ``.emscripten`` file).
+.. note:: The ``WASM``, ``BINARYEN*``, etc. options only matter when compiling to your final executable. In other words, the same .o files are used for both asm.js and WebAssembly. Only when linking them and compiling to asm.js or WebAssembly do you need to specify WebAssembly if you want that. That means that it is easy to build your project to both asm.js and WebAssembly. (The only exception is the wasm backend's ``WASM_OBJECT_FILES`` flag which determines whether object files contain wasm or LLVM IR.)
```

Comment:
Are we gonna switch `WASM_OBJECT_FILES = 1` by default soon? If so this paragraph may not be valid anymore?

Reply:
We already did that switch a while ago (we decided to do it before switching the backend, so people have just one big change instead of 2). Then yeah, as you said, this paragraph is not valid... I'll rewrite.